### ISSUE TITLE
Add ability to specify channel polling pattern 

### ISSUE THREAD

**Issue by ktf** [FairRootGroup/FairMQ#58]
I discussed this with Mohammad, putting it here to start a discussion.

As we discussed a while ago, I implement HandleData(), but do not process data immediately, for example waiting for extra inputs of a computation. This means that if I have two devices pushing to a third one, this can result in an imbalance in the queues with all of its consequences.

I would therefore like to have a way to rescale the polling on a channel with respect to another, so that I can reduce the rate at which I get messaged on the fast one. Is that possible? Is there a better alternative?

**Comment by dennisklein** [comment]
`OnData` is less suitable in your case, you can instead override `FairMQDevice::Run` or `FairMQDevice::ConditionalRun` and iterate over your input channels and call `FairMQChannel::Receive` (or similar on each channel, there is a strictly blocking `Receive(msg)`, a blocking `Receive(msg, timeout)`, and a non-blocking `ReceiveAsync(msg)` (evaluate return value here) - the blocking ones may be cancelled on device reset though, see return value for distinguishing) on each channel.

`virtual bool FairMQDevice::ConditionalRun()` is called in a loop during RUN state until it returns `false` or if device transitioned out of RUN. See [link](https://github.com/FairRootGroup/FairMQ/blob/master/fairmq/FairMQDevice.cxx#L497-L511).

With `virtual void FairMQDevice::Run()` you need to write a loop with the check for device state change yourself, if you want to run indefinitely otherwise.

To summarize, we recommend something like this:

```cpp
struct MyDevice : FairMQDevice {
// ...
  bool MyDevice::ConditionalRun override { // gets called repeatedly
    for (const auto& channel : myInputChannels) {
      FairMQMessage msg; // better created with NewMessageFor() factory
      channel.Receive(msg);
      // ...
    }
    return true;
  }
// ...
}
```

**Comment by ktf** [comment]
Yeah, I was just thinking I could have done it directly with HandleData, but in the end you are right, better to simply do it myself and use `Run`.
